lib/mastodon/email_domains_cli.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require_relative '../../config/boot'
+require_relative '../../config/environment'
+require_relative 'cli_helper'
+
+module Mastodon
+  class EmailDomainsCLI < Thor
+    include CLIHelper
+
+    def self.exit_on_failure?
+      true
+    end
+
+    desc 'list', 'list E-mail domain blocks'
+    long_desc <<-LONG_DESC
+      list up all E-mail domain blocks.
+    LONG_DESC
+    def list
+      EmailDomainBlock.where(parent_id: nil).includes(:children).order(id: 'DESC').find_each do |entry|
+        say(entry.domain.to_s, :green)
+        entry.children.order(id: 'DESC').find_each do |child|
+          say("  #{child.domain}", :yellow)
+        end
+      end
+    end
+
+    option :with_dns_records, type: :boolean
+    desc 'block [DOMAIN...]', 'Block E-mail domains'
+    long_desc <<-LONG_DESC
+      Block E-mail domains from a given DOMAIN.
+
+      When the --with-dns-records option is given, An attempt to resolve the
+      given domain's DNS records will be made and the results will also be
+      blacklisted.
+    LONG_DESC
+    def block(*domains)
+      if domains.empty?
+        say('No domain(s) given', :red)
+        exit(1)
+      end
+
+      domains.each do |domain|
+        if EmailDomainBlock.where(domain: domain).exists?
+          say("#{domain} is already blocked.", :yellow)
+          next
+        end
+
+        email_domain_block = EmailDomainBlock.new(domain: domain, with_dns_records: options[:with_dns_records] || false)
+        email_domain_block.save!
+
+        say("#{domain} was blocked.", :green)
```

Comment:
In other CLI commands, we just count up number of successes and then return a message at the end like "Blocked 123 domains"